src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -369,10 +375,15 @@ public static void checkApiUrlValidity(@Nonnull GitHub gitHub, @CheckForNull Sta
 
                 GitHubBuilder gb = createGitHubBuilder(apiUrl, cache);
 
-                if (username != null) {
-                    gb.withPassword(username, password);
+                if (gitHubAppCredentials != null) {
+                    gb.withAuthorizationProvider(gitHubAppCredentials.getAuthorizationProvider());
+                } else if (username != null && password != null) {
+                    // At the time of this change this works for OAuth tokens as well.
+                    // This may not continue to work in the future, as GitHub has deprecated Login/Password credentials. 
```

Comment:
at some point should really switch to SecretText credentials so you can easily tell the difference...